src/shogun/lib/SGMatrix.h

```diff
@@ -309,6 +333,68 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** fill matrix with zeros */
 		void zero();
 
+		/**
+		 * fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		 * Rest: disabled
+		**/
+		// vectorized available
+		template <typename U>
+		using enable_simd_float = std::enable_if_t<is_any<U, float64_t>::value>;
+		template <typename U = T>
+		auto random() -> enable_simd_float<U>
+		{
+			CRandom r{};
+			r.fill_array_co(
+			    static_cast<float64_t*>(matrix), num_rows * num_cols);
+		}
+
+		template <typename U>
+		using enable_simd_int =
+		    std::enable_if_t<is_any<U, uint32_t, uint64_t>::value>;
+		template <typename U = T>
+		auto random() -> enable_simd_int<U>
+		{
+			CRandom r{};
+			r.fill_array(matrix, num_rows * num_cols);
+		}
+
+		// vectorized not available
+		template <typename U>
+		using enable_nonsimd_float =
+		    std::enable_if_t<is_any<U, float32_t, floatmax_t>::value>;
+		template <typename U = T>
+		auto random() -> enable_nonsimd_float<U>
+		{
+			CRandom r{};
+			// Casting floats upwards or downwards is safe
+			// Ref: https://stackoverflow.com/a/36840390/3656081
+			for (index_t i = 0; i < num_rows * num_cols; i++)
```

Comment:
std::for_each ?

Reply:
`std::for_each` wont work with a stateful lambda which would be needed here `[&r]{...r.random...}`?

Reply:
I just tried this out on my machine, and both methods seem to take the same amount of time, I am not sure there is a way to speed up either methods.

Reply:
And btw, I wouldn't recommend this, but you could do this with `std::for_each`:
```
std::for_each(matrix, matrix+N, 
	[&r, &matrix](auto a){
		static int i = 0; 
		matrix[i] = r.random_half_open(); 
		++i; 
	});
```

Reply:
any thoughts on this @saatvikshah1994 ?

Reply:
Yep, std::transform is indeed equally fast(or even faster at times especially at lower optimization levels). This is done with `std::transform` - the `for_each` version looked a bit hacky :/